assets/js/blocks/product-categories/frontend.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { render } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Block from './block.js';
+
+const containers = document.querySelectorAll(
+	'.wp-block-woocommerce-product-categories'
+);
+
+if ( containers.length ) {
+	containers.forEach( ( el ) => {
```

Comment:
This a nifty way of handling this 👍 

Obviously beyond the scope of this PR, but if this does become the standard for frontend blocks, it would be cool to have a method that can automatically detect and pull attributes from a given class for any given block.